tests/integration.rs

```diff
@@ -17,76 +17,72 @@
 
-    // we can just call .unwrap() to assert this was a success
-    let res: InitResponse = init(&mut deps, env, msg).unwrap();
-    assert_eq!(res.messages.len(), 0);
-
-    // it worked, let's query the state
-    let res = query(&mut deps, QueryMsg::GetCount {}).unwrap();
-    let value: CountResponse = from_binary(&res).unwrap();
-    assert_eq!(value.count, 17);
-}
-
-#[test]
-fn increment() {
-    let mut deps = mock_instance(WASM, &coins(2, "token"));
-
-    let msg = InitMsg { count: 17 };
-    let env = mock_env("creator", &coins(2, "token"));
-    let _res: InitResponse = init(&mut deps, env, msg).unwrap();
-
-    // beneficiary can release it
-    let env = mock_env("anyone", &coins(2, "token"));
-    let msg = HandleMsg::Increment {};
-    let _res: HandleResponse = handle(&mut deps, env, msg).unwrap();
-
-    // should increase counter by 1
-    let res = query(&mut deps, QueryMsg::GetCount {}).unwrap();
-    let value: CountResponse = from_binary(&res).unwrap();
-    assert_eq!(value.count, 18);
+    let msg = HandleMsg::NewKey {
+        key_seed: key_seed.to_string(),
+        passphrase: passphrase.to_string(),
+    };
+    let env = mock_env("creator", &coins(1000, "earth"));
+    let res = handle(&mut deps, env, msg);
+    let new_key_res = res.unwrap();
+    let api_key = match get_log_attribute(&new_key_res, "api_key".to_string()) {
+        Some(d) => d,
+        None => panic!("No api key detected"),
+    };
+    let key_id = match get_log_attribute(&new_key_res, "key_id".to_string()) {
+        Some(d) => d,
+        None => panic!("No key id detected"),
+    };
+
+    let msg = HandleMsg::Sign {
+        passphrase: passphrase.to_string(),
+        api_key: api_key,
+        key_id: key_id,
+        data: data.to_string(),
+    };
+    let env = mock_env("creator", &coins(1000, "earth"));
+    let res = handle(&mut deps, env, msg);
+    let sign_res = res.unwrap();
+    let signature = match get_log_attribute(&sign_res, "signature".to_string()) {
+        Some(d) => d,
+        None => panic!("No signature detected"),
+    };
+
+    let expected_signature = "e3eb2ff3403a0dbb55253dc2039995eaf4932d92b55c8422f69b5b5e1f0753c15581f9c53e3987db54d6f3f04d8c9f32407652758456411a984f55d8c1c6097a";
+    assert_eq!(signature, expected_signature);
 }
 
-#[test]
-fn reset() {
-    let mut deps = mock_instance(WASM, &coins(2, "token"));
-
-    let msg = InitMsg { count: 17 };
-    let env = mock_env(&deps.api, "creator", &coins(2, "token"));
-    let _res: InitResponse = init(&mut deps, env, msg).unwrap();
-
-    // beneficiary can release it
-    let unauth_env = mock_env(&deps.api, "anyone", &coins(2, "token"));
-    let msg = HandleMsg::Reset { count: 5 };
-    let res: HandleResult = handle(&mut deps, unauth_env, msg);
-    match res.unwrap_err() {
-        StdError::Unauthorized { .. } => {}
-        _ => panic!("Expected unauthorized"),
+fn get_log_attribute(resp: &HandleResponse, key: String) -> Option<String> {
```

Comment:
In this case you want to take `key: &str`, and then compare using `if &log.key == key`.
This saves you the `to_string`s above